syntax = "proto3";

package eventmaster;

// Define the event store API
service EventMaster {
    rpc FireEvent (Event) returns (WriteResponse) {}
    rpc GetEvents (Query) returns (stream Event) {}
}

message Event {
    string event_id = 1;
    string parent_event_id = 2;
    int64 event_time = 3;
    string dc = 4;
    string topic_name = 5;
    repeated string tag_set = 6;
    string host = 7;
    repeated string target_host_set = 8;
    string user = 9;
    string data = 10;
}

message WriteResponse {
    int32 errcode = 1;
    string errmsg = 2;
    string event_id = 3;
}

message Query {
    repeated string dc = 1;
    repeated string host = 2;
    repeated string target_host = 3;
    repeated string topic_name = 4;
    repeated string tag_set = 5;
    string data = 6;
    int64 start_event_time = 7;
    int64 end_event_time = 8;
    int64 start_received_time = 9;
    int64 end_received_time = 10;
    repeated string sort_field = 11;
    repeated bool sort_ascending = 12;
    int32 start = 13;
    int32 limit = 14;
}

message HealthcheckRequest {}

message HealthcheckResponse {
    string response = 1;
}
